### ISSUE TITLE
[Craft 4]: Rendering issue with Group field

### ISSUE THREAD

**Issue by JeanLucEsser** [verbb/formie#879]
**Description**

The label for a Group is misplaced.
A photo being worth a thousands words... see attached!

![](https://cdn.discordapp.com/attachments/575387375610626063/973129946732044298/unknown.png)

**Comment by engram-design** [comment]
I'll say that this is _sort of_ not a bug, but can be defaulted better. The tricky thing with this structure (for accessibility and semantics) is that it's modelled as a `<fieldset>` and `<legend>` (the field label being the legend). As such, the legend should be placed within the fieldset. There's all sorts of weird position things we can do to move it visually outside of the fieldset container, but none I'm really that keen on as defaults. You're obviously free to change this as required. In fact, you can't really even position the label/legend anywhere else apart from the top - just a HTML thing I believe.

As for the instruction text, we're not bound to the same requirements as the label, needing to be inside the fieldset. The default behaviour is (regrettably) to put it "Above Input" which puts it outside the fieldset. 99% of the time, you'll likely want to pick "Top of Fieldset" which would display it under the legend, as you'd expect.

I'm sure there's scenarios where the instruction text would want to be shown outside the fieldset, and the label hidden, but we can probably look at cleaning these positions up for fieldset-enabled fields.

**Comment by JeanLucEsser** [comment]
I sort of get it ;)
The funny thing is you render it as it should in the form designer but I get you’re not dealing with the same constraints. 

Maybe I’m wrong but how about defaulting to label and instructions inside the fieldset, and not styling the fieldset itself, but styling an inner container containing all the fields?

**Comment by engram-design** [comment]
Yeah, I'll admit it's not handled gracefully in the preview, leading to unexpected layouts right now.

I'll consider that, and it's probably the best approach (wrapper around fields, as opposed to styling the fieldset). I didn't really consider it at the time when I wrote the templates on the implications/limitations of the structure of it. I'll give it some more thought along with [upcoming template refactoring](https://github.com/verbb/formie/issues/877#issuecomment-1124933028)

At the very least, I'll modify the defaults to be placed inside the fieldset

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.0-beta.16",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.0](https://github.com/verbb/formie/releases/tag/2.0.0)
